### PR TITLE
[CALCITE-2993] ParseException may be thrown for legal SQL queries due…

### PR DESCRIPTION
**Pull request by zhztheplayer**
… to incorrect "LOOKAHEAD(1)" hints

Also:
* Following [CALCITE-883], add absent non-reserved keywords to config.fmpp files;
* Remove redundant "LOOKAHEAD(1)" hints;
* The case "select json_object(key: value) from t" is added to unit test but not fixed. See the reason from method "SqlParserTest#testJsonObject".

### CODE REVIEW

File: core/src/main/codegen/templates/Parser.jj [hunk 1]
```diff
@@ -1894,7 +1894,7 @@ SqlNode TableRef2(boolean lateral) :
             tableRef = unnestOp.createCall(s.end(this), args.toArray());
         }
     |
-        [ LOOKAHEAD(1) <LATERAL> { lateral = true; } ]
```
Comment (zhztheplayer) [inline core/src/main/codegen/templates/Parser.jj]: This one is removed cause it is redundant:
1. For core parsers (global lookahead = 1), the `LOOKAHEAD(1)` is already implied;
2. For babel parser (global lookahead = 2), `[LOOKAHEAD(1) <FOO>]` is not faster than `[LOOKAHEAD(2) <FOO>]`. They are basically equivalent.

File: core/src/main/codegen/templates/Parser.jj [hunk 2]
```diff
@@ -3422,7 +3429,6 @@ SqlNode AtomicRowExpression() :
 }
 {
     (
-        LOOKAHEAD(1)
```
Comment (zhztheplayer) [inline core/src/main/codegen/templates/Parser.jj]: This one is removed cause it is redundant. (Same reason, production `Literal()` contains only a single token)

**Comment by chunweilei** [comment]
LGTM, +1.

**Comment by chunweilei** [comment]
BTW, does it not cause build broken if some non-reserved keywords are not added to `config.fmpp`?

**Comment by zhztheplayer** [comment]
Thank you very much for the review, @chunweilei !

Address your comment:
> Does it not cause build broken if some non-reserved keywords are not added to `config.fmpp`?

No currently it doesn't. If you meant automating the build to correct the inconsistency of non-reserved keyword list among different parsers, I'll support this idea. At least at this time, the default parser's non-reserved keywords are all added to other parsers, and it ensures that if a query is parsed successfully by core parser, it should also be safely parsed by other parsers.

**Pull request merged**